src/HDF5_MultiDim.chpl

```diff
@@ -0,0 +1,409 @@
+                           errorClass="IllegalArgumentError");
+            }
+        }
+
+        // write the dataset based in the format provided
+        var rank: c_int;
+        select (method) {
+            when (FLAT) { // store data as a flattened single array
+                rank = 1:c_int;
+                var dims: [0..#1] C_HDF5.hsize_t;
+                dims[0] = (* reduce shape.a):C_HDF5.hsize_t;
+                C_HDF5.H5LTmake_dataset(file_id, dset_name.c_str(), rank, dims, dtype_id, A.ptr);
+            }
+            when (MULTI_DIM) { // store the data as the multi-dimensional object
+                rank = shape.size:c_int;
+                // reshape the data based on the shape passed.
+                var dims: [0..#shape.size] C_HDF5.hsize_t;
+                forall i in 0..#shape.size{
+                    dims[i] = shape.a[i]:C_HDF5.hsize_t;
+                }
+                C_HDF5.H5LTmake_dataset(file_id, dset_name.c_str(), rank, dims, dtype_id, A.ptr);
+            }
+            otherwise {
+                throw getErrorWithContext(
+                           msg="Unknown storage method. Expecting 0 (flat) or 1 (multi). Found %s".format(method),
+                           lineNumber=getLineNumber(),
+                           routineName=getRoutineName(), 
+                           moduleName=getModuleName(),
+                           errorClass="IllegalArgumentError");
+            }
+        }
+        
+    }
+
+    /*
+    * Takes a multidimensional array obj and writes it into and HDF5 dataset.
+    * Provides the ability to store the data flat or multidimensional.
+    */
+    proc write_hdf_multi_msg(cmd: string, payload: string, st: borrowed SymTab): MsgTuple throws {
+        var (flat_name, shape_name, order_str, filename, dset_name, mode_str, method_str) = payload.splitMsgToTuple(7);
+        
+        var method: int;
+        try {
+            method = method_str:int;
+        } catch {
+            var errorMsg = "Could not convert method to numeric";
+            h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR);
+        }
+
+        var entry = st.lookup(flat_name);
+        var entryDtype = DType.UNDEF;
+        if (entry.isAssignableTo(SymbolEntryType.TypedArraySymEntry)) {
+            entryDtype = (entry: borrowed GenSymEntry).dtype;
+        } else if (entry.isAssignableTo(SymbolEntryType.SegStringSymEntry)) {
+            entryDtype = (entry: borrowed SegStringSymEntry).dtype;
+        } else {
+            var errorMsg = "writehdf_multi Unsupported SymbolEntryType:%t".format(entry.entryType);
+            h5tLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR);
+        }
+
+        var flat_sym: borrowed GenSymEntry = getGenericTypedArrayEntry(flat_name, st);
+        // var flat = toSymEntry(flat_sym, int);
```

Comment:
It doesn't look like flat_sym is used anywhere. Unless I'm just missing it, these two lines could probably be pulled out

Reply:
You are correct. I changed the code structure and forgot to remove this. Updated.